stats/stats.go

```diff
@@ -0,0 +1,86 @@
+package stats
+
+import (
+	"github.com/ozontech/file.d/logger"
+	"github.com/ozontech/file.d/pipeline"
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+type MetricDesc struct {
+	Subsystem string
+	Name      string
+	Help      string
+}
+
+type key struct {
+	namespace string
+	subsystem string
+	name      string
+}
+
+type AnyMetric prom.Collector
+
+type stats struct {
+	allMetrics map[key]AnyMetric
+}
+
+const (
+	subsystemName  = "stats"
+	unknownCounter = "unknown_counter"
+)
+
+var statsGlobal *stats
+
+func InitStats() {
+	statsGlobal = &stats{
+		allMetrics: make(map[key]AnyMetric),
+	}
+
+	statsGlobal.registerUnknown()
+}
+
+func RegisterCounter(metricDesc *MetricDesc) {
+	maskPromCounter := prom.NewCounter(prom.CounterOpts{
+		Namespace: pipeline.PromNamespace,
+		Subsystem: metricDesc.Subsystem,
+		Name:      metricDesc.Name,
+		Help:      metricDesc.Help,
+	})
+
+	keyInternal := key{pipeline.PromNamespace, metricDesc.Subsystem, metricDesc.Name}
+	registerMetric(keyInternal, maskPromCounter)
+}
+
+func GetCounter(subsystem, metricName string) prom.Counter {
+	if statsGlobal == nil {
+		logger.Panicf("stats package uninitialized")
```

Comment:
Is it responsibility of GetCounter? Probably better return error and resolve on higher level.

Reply:
Another option is initialization in init function in stats package.

Reply:
@anuriq @vano144 should we leave this panic here or remove it?